Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -797,4 +797,47 @@ def test_custom_command
         cmd(cmd, {"PATH" => "#{path}#{File::PATH_SEPARATOR}#{ENV["PATH"]}"})
     end
   end
+
+  def test_search
+    formula_file = CoreTap.new.formula_dir/"testball.rb"
+    formula_file.write <<-EOS.undent
+      class Testball < Formula
+        desc "Some test"
+        url "https://example.com/testball-0.1.tar.gz"
+      end
+    EOS
+
+    refute_predicate HOMEBREW_CACHE.join("desc_cache.json"),
+                     :exist?, "Cached file should not exist"
+
+    assert_match "testball", cmd("search")
+    assert_match "testball", cmd("search", "testball")
+    assert_match "testball", cmd("search", "homebrew\/homebrew-core\/testball")
+    assert_match "testball", cmd("search", "--desc", "Some test")
+
+    flags = Hash.new
```

Comment:
You can initialize an empty `Hash` instance simply by using `{}` (equivalent to `Hash.new`, but shorter and generally preferred). Better yet, to avoid having to spell out `flags` for every assignment and construct the entire hash with all its entries at once, do something like:

```ruby
flags = {
  "macports" => "https://www.macports.org/ports.php?by=name&substr=testball",
  "fink" => "http://pdb.finkproject.org/pdb/browse.php?summary=testball",
  # …
}
```